data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumer.java

```diff
@@ -424,16 +426,25 @@ private void processRecord(final AcknowledgementSet acknowledgementSet, final Re
                 bufferAccumulator.add(record);
                 break;
             } catch (Exception e) {
+                if (!paused) {
+                    paused = true;
```

Comment:
Should kafka consumer be paused for all exception ?

Reply:
In all cases of exception, we are trying for ever to put in the buffer, right? that's why we are pausing in all cases of exceptions.